src/Compilers/CSharp/Portable/Symbols/LocalDeclarationKind.cs

```diff
@@ -49,5 +49,16 @@ internal enum LocalDeclarationKind : byte
         /// The variable that captures the result of a pattern matching operation like "i" in "expr is int i"
         /// </summary>
         PatternVariable,
+
+        /// <summary>
+        /// User variable declared by a declaration expression in the left-hand-side of a deconstruction assignment.
+        /// </summary>
+        Deconstruction,
+
+        /// <summary>
+        /// User variable declared by a declaration expression that is not a deconstruction. This occurs for an
+        /// out variable declaration, and as a result of error recovery in incorrect code.
+        /// </summary>
+        DeclarationExpression,
```

Comment:
Looking at this line specifically

>  This occurs for an out variable declaration, and as a result of error recovery in incorrect code.

Having `DeclarationExpression` represent declaration expressions and error recovery doesn't seem right to me.  I can't think of a good justification for having error scenarios default to `DeclarationExpression` vs. any of the other values.  Why don't we have an explicit error case here?